packages/design-system/src/components/OcBreadcrumb/OcBreadcrumb.vue

```diff
@@ -142,6 +181,69 @@ export default defineComponent({
   },
   setup(props) {
     const { $gettext } = useGettext()
+    const visibleItems = ref<BreadcrumbItem[]>([])
+    const hiddenItems = ref<BreadcrumbItem[]>([])
+    const displayItems = ref<BreadcrumbItem[]>(props.items.slice())
+
+    const getBreadcrumbElement = (id): HTMLElement => {
+      return document.querySelector(`[data-item-id="${id}"]`)
+    }
+
+    const calculateTotalBreadcrumbWidth = () => {
+      let totalBreadcrumbWidth = 100 // 100px margin to the right to avoid breadcrumb from getting too close to the controls
+      visibleItems.value.forEach((item, index) => {
+        const breadcrumbElement = getBreadcrumbElement(item.id)
+        const itemClientWidth = breadcrumbElement?.getBoundingClientRect()?.width || 0
+        totalBreadcrumbWidth += itemClientWidth
+      })
+      return totalBreadcrumbWidth
+    }
+
+    const reduceBreadcrumb = (offsetIndex) => {
+      const breadcrumbMaxWidth = props.maxWidth
+      if (!breadcrumbMaxWidth) {
+        return
+      }
+      document.getElementById(props.id)?.style.setProperty('--max-width', `${breadcrumbMaxWidth}px`)
+      const totalBreadcrumbWidth = calculateTotalBreadcrumbWidth()
+
+      const isOverflowing = breadcrumbMaxWidth < totalBreadcrumbWidth
+      if (!isOverflowing || visibleItems.value.length <= 3) {
+        return
+      }
+      // Remove from the left side
+      const removed = visibleItems.value.splice(offsetIndex, 1)
+
+      hiddenItems.value.push(removed[0])
+      reduceBreadcrumb(offsetIndex)
+    }
+
+    const lastHiddenItem = computed(() =>
+      hiddenItems.value.length >= 1 ? unref(hiddenItems)[unref(hiddenItems).length - 1] : { to: {} }
+    )
+
+    const renderBreadcrumb = () => {
+      displayItems.value = props.items.slice()
+      if (displayItems.value.length > 1) {
+        displayItems.value.splice(1, 0, {
+          text: '...',
+          allowContextActions: false,
+          to: {},
+          isTruncated: true
+        })
+      }
+      visibleItems.value = displayItems.value.slice()
+      hiddenItems.value = []
+
+      nextTick(() => {
+        reduceBreadcrumb(2)
```

Comment:
Please introduce a param for the offset. Don't forget to use it when inserting the `...` element to the correct position. Use `2` as default for the offset param. But then set `3` for breadcrumbs in project spaces. That way you get `Spaces > SpaceName > ... > trölölö` for project spaces. Otherwise the space name will always be cut off.

Reply:
yeah i implemented it BUT, as soon as i navigate into a space im no longer in a project space but in a generic space, the same as personal

Reply:
But you do have a `space` object via prop in the generic space view. And on that `space` object you can use the type guard `isProjectSpaceResource(props.space)` to identify if it's a project space or not. I think you also need the same with `isShareSpaceResource(props.space)` because a share breadcrumb also has an overview item and a share root item in the breadcrumbs (`Shared with me > some share name > ... > subfolder`).